decimal.go

```diff
@@ -81,21 +81,28 @@ func New(value int64, exp int32) Decimal {
 //
 func NewFromString(value string) (Decimal, error) {
 	var intString string
-	var exp int32
+	var exp int64
+
+	// Check if number is using scientific notation
+	eIndex := strings.IndexAny(value, "Ee")
+	if eIndex != -1 {
+		expInt, err := strconv.ParseInt(value[eIndex+1:len(value)], 10, 64)
```

Comment:
Agh, I just noticed this: if `expInt == math.MinInt64`, the line where you `exp += int64(expInt)` will cause an integer overflow. You should parse this as a 32 bit int, and cast it to an int64 when you assign to `exp` below.